pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -677,7 +677,16 @@ func (a *ApplicationRestoreController) restoreVolumes(restore *storkapi.Applicat
 					return err
 				}
 				var preRestoreObjects []runtime.Unstructured
-				if !nfs {
+				if nfs {
+					if restore.Spec.ReplacePolicy == storkapi.ApplicationRestoreReplacePolicyRetain {
```

Comment:
Let us please add comment explaining, why we are doing this. 
Also I prefer to have the check as "if nfs && driverName == volume.PortworxDriverName" to be clear, Even though current check will work.

Reply:
> Let us please add comment explaining, why we are doing this. Also I prefer to have the check as "if nfs && driverName == volume.PortworxDriverName" to be clear, Even though current check will work.

Added this thing as comment